### PR TITLE
Fix filtering top repositories (#783, #1281)

### LINKED ISSUE #783: Harbor should not display private repositories in the home page - 2ed
Hello,

IMHO, [issue 621](https://github.com/vmware/harbor/issues/621) is not fully fixed; private repository names are still displayed in the public home page (tested in 0.3.5).

More generally, the private elements of this list should not be obtained anonymously via the API (curl -v  https://SERVER/api/repositories/top). 

It would be wiser to not display anything about private repositories as their names can contain indications on internal (=private) entreprise activities.

### PR DESCRIPTION
**Pull request by laz2**
Fix #783, #1281:
- filter repositories from deleted projects
- filter repositories from invisible projects

### CODE REVIEW

**Comment by yauhen-l** [comment]
+1

**Comment by amelius0712** [comment]
+1

**Comment by andreikop** [comment]
+1

File: src/ui/api/repository.go [hunk 1]
```diff
@@ -416,7 +416,7 @@ func (ra *RepositoryAPI) GetTopRepos() {
 		ra.CustomAbort(http.StatusBadRequest, "invalid count")
 	}
 
-	repos, err := dao.GetTopRepos(count)
+	repos, err := dao.GetTopRepos(ra.GetSession("userId"), count)
```
Comment (ywk253100) [inline src/ui/api/repository.go]: Use function GetUserIDForRequest in src/common/api/base.go to get ID of user. It supports authenticating user against both basic auth and session.
BTW, open the PR to branch dev please.

**Comment by laz2** [comment]
Hmmm, i think this is not my changes fail tests. Or not?

**Comment by ywk253100** [comment]
@laz2 Open a new PR to branch dev please, there may be some issue on travis, it can not get the latest travis.yml.

**Comment by laz2** [comment]
Need recreating for dev branch

**Pull request closed without merging**